recipe-client-addon/content/AboutPages.jsm

```diff
@@ -0,0 +1,208 @@
+const PROCESS_SCRIPT = (
+  `resource://shield-recipe-client/content/shield-content-process.js?${Math.random()}`
+);
+const FRAME_SCRIPT = (
+  `resource://shield-recipe-client/content/shield-content-frame.js?${Math.random()}`
+);
+
+/**
+ * Class for managing an about: page that Shield provides. Adapted from
+ * browser/extensions/pocket/content/AboutPocket.jsm.
+ *
+ * @implements nsIFactory
+ * @implements nsIAboutModule
+ */
+class AboutPage {
+  constructor({chromeUrl, aboutHost, classId, description, uriFlags}) {
+    this.chromeUrl = chromeUrl;
+    this.aboutHost = aboutHost;
+    this.classId = Components.ID(classId);
+    this.description = description;
+    this.uriFlags = uriFlags;
+  }
+
+  getURIFlags() {
+    return this.uriFlags;
+  }
+
+  newChannel(uri, loadInfo) {
+    const newURI = Services.io.newURI(this.chromeUrl);
+    const channel = Services.io.newChannelFromURIWithLoadInfo(newURI, loadInfo);
+    channel.originalURI = uri;
+
+    if (this.uriFlags & Ci.nsIAboutModule.URI_SAFE_FOR_UNTRUSTED_CONTENT) {
+      const principal = Services.scriptSecurityManager.createCodebasePrincipal(uri, {});
+      channel.owner = principal;
+    }
+    return channel;
+  }
+
+  createInstance(outer, iid) {
+    if (outer !== null) {
+      throw Cr.NS_ERROR_NO_AGGREGATION;
+    }
+    return this.QueryInterface(iid);
+  }
+
+  /**
+   * Register this about: page with XPCOM. This must be called once in each
+   * process (parent and content) to correctly initialize the page.
+   */
+  register() {
+    Cm.QueryInterface(Ci.nsIComponentRegistrar).registerFactory(
+      this.classId,
+      this.description,
+      `@mozilla.org/network/protocol/about;1?what=${this.aboutHost}`,
+      this,
+    );
+  }
+
+  /**
+   * Unregister this about: page with XPCOM. This must be called before the
+   * add-on is cleaned up if the page has been registered.
+   */
+  unregister() {
+    Cm.QueryInterface(Ci.nsIComponentRegistrar).unregisterFactory(this.classId, this);
+  }
+}
+AboutPage.prototype.QueryInterface = XPCOMUtils.generateQI([Ci.nsIAboutModule]);
+
+/**
+ * The module exported by this file.
+ */
+this.AboutPages = {
+  async init() {
+    // Load scripts in content processes and tabs
+    Services.ppmm.loadProcessScript(PROCESS_SCRIPT, true);
+    Services.mm.loadFrameScript(FRAME_SCRIPT, true);
+
+    // Register about: pages and their listeners
+    this.aboutStudies.register();
+    this.aboutStudies.registerParentListeners();
+
+    CleanupManager.addCleanupHandler(() => {
+      // Stop loading processs scripts and notify existing scripts to clean up.
+      Services.ppmm.removeDelayedProcessScript(PROCESS_SCRIPT);
+      Services.ppmm.broadcastAsyncMessage("Shield:ShuttingDown");
+      Services.mm.removeDelayedFrameScript(FRAME_SCRIPT);
+      Services.mm.broadcastAsyncMessage("Shield:ShuttingDown");
+
+      // Clean up about pages
+      this.aboutStudies.unregisterParentListeners();
+      this.aboutStudies.unregister();
```

Comment:
This is called twice, I think? Once here and once from the process script. I guess we should only add this to the cleanup manager if we're in the parent process, or something - and the content process script, is that only loaded in the content process, or also in the parent? Maybe that needs an exception, too, or we delegate the about page stuff to one of the two...